src/parser/tests/expression/NullCoalescenceExpression.spec.ts

```diff
@@ -374,5 +380,174 @@ describe('NullCoalescingExpression', () => {
                 end sub
             `);
         });
+
+        it('transpiles null coalescing in RHS of AssignmentStatement to if statement', () => {
+            testTranspile(`
+                sub main()
+                    a = user ?? {}
+                end sub
+            `, `
+                sub main()
+                    a = user
+                    if a = invalid then
+                        a = {}
+                    end if
+                end sub
+            `);
+        });
+
+        it('transpiles null coalescing in RHS of incrementor AssignmentStatement to if statement', () => {
+            testTranspile(`
+                sub main()
+                    a += user ?? 0
+                end sub
+            `, `
+                sub main()
+                    a += user
+                    if a = invalid then
+                        a += 0
+                    end if
+                end sub
+            `);
+        });
+
+        it('transpiles null coalescing in RHS of DottedSetStatement to if statement', () => {
+            testTranspile(`
+                sub main()
+                    m.a = user ?? {}
+                end sub
+            `, `
+                sub main()
+                    m.a = user
+                    if m.a = invalid then
+                        m.a = {}
+                    end if
+                end sub
+            `);
+        });
+
+        it('transpiles null coalescing in RHS of incrementor DottedSetStatement to if statement', () => {
+            testTranspile(`
+                sub main()
+                    m.a += user ?? 0
+                end sub
+            `, `
+                sub main()
+                    m.a += user
+                    if m.a = invalid then
+                        m.a += 0
+                    end if
+                end sub
+            `);
```

Comment:
This one is also incorrect (see previous comment). must be handled like `m.a += (user ?? 0)`